File: C/factorial.c
#include <stdio.h>
int main() {
    int n, i;
    unsigned long long fact = 1;
l1:    printf("Enter an integer for which you want to find factorial: ");
    scanf("%d", &n);

    // shows error if the user enters a negative integer
    if(n<0){
        printf("Factorial of negative number does not exist\nEnter positive number.\n");
        goto l1;
    }
    else{
        for(i=1;i<n;++i){
            fact*=i;
        }
    }
    printf("Factorial of %d = %llu", n, fact);
    return 0;
}

File: Java/Factorial.java
import java.util.Scanner;

/**
 * Factorial
 */
public class Factorial {

    static int factorial(int a) {
        if (a==0 || a==1) {
            return 1;
        }
        else {
            return factorial(a-1)*a;
        }
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.println("Enter a number: ");
        System.out.println("Factorial = "+factorial(in.nextInt())); 
        in.close();
    }
}
